fs/fs.go

```diff
@@ -187,10 +192,44 @@ type sociContext struct {
 	imageLayerToSociDesc map[string]ocispec.Descriptor
 }
 
-func (c *sociContext) Init(ctx context.Context, imageRef, indexDigest string, store orascontent.Storage) error {
+func (c *sociContext) Init(ctx context.Context, imageRef, indexDigest, imageDigest string, store orascontent.Storage) error {
```

Comment:
nit: can we rename `imageDigest` to `imageManifestDigest` to be clear that we're talking about a specific iamge manifest and not an image index?